Applications/utils/mem_usage.sh

```diff
@@ -18,10 +18,19 @@ count=0
 while true; do
     if [ -d "/proc/$process_id" ]; then
 	output=`grep VmHWM /proc/$process_id/status`
-	IFS=' ' read -ra mem_str <<< "$output"
+	read -ra mem_str <<< "$output"
 	count=$((count + 1))
 	mem_usage=$((${mem_str[1]}))
-	outlog="$outlog$count $mem_usage kB\n"
+	cpu_output=""
+	n=`nproc`
+	i=0
+	while [ "$i" -lt $n ]
+	do
+	    cpu_freq=`cat /sys/devices/system/cpu/cpu$i/cpufreq/scaling_cur_freq`
+	    cpu_output="$cpu_output $cpu_freq"
+	    i=$(expr $i + 1)
+	done
+	outlog="$outlog$count $mem_usage kB $cpu_output\n"
```

Comment:
If you want to get more info on CPU usages... than simply counting wallclock time / cpu-time & cpu-freq, you may use perf (e.g., ```$ perf stat -- nntrainer_test_case ```)

Reply:
It will give you something like:
```
Performance counter stats for 'nntrainer_test_case':

  244011.782059  task-clock-msecs         #      0.925 CPUs 
          53328  context-switches         #      0.000 M/sec
            515  CPU-migrations           #      0.000 M/sec
        1843121  page-faults              #      0.008 M/sec
   789702529782  cycles                   #   3236.330 M/sec
  1050912611378  instructions             #      1.331 IPC  
   275538938708  branches                 #   1129.203 M/sec
     2888756216  branch-misses            #      1.048 %    
     4343060367  cache-references         #     17.799 M/sec
      428257037  cache-misses             #      1.755 M/sec

  263.779192511  seconds time elapsed
```

Reply:
Thanks for the tip. One of the reason logging the CPU frequency is to show CPU throttling.

Reply:
This looks beautiful! (and insightful!)